src/rum/config.cpp

```diff
@@ -0,0 +1,154 @@
+#include "config.h"
+
+#include <rapidjson/document.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+#include "string_util.h"
+
+namespace {
+
+char *set_config(ngx_conf_t *cf, ngx_command_t *command, void *conf) {
+  auto *rum_config =
+      static_cast<std::unordered_map<std::string, std::string> *>(conf);
+
+  if (cf->args->nelts < 2) {
+    char *err_msg =
+        static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
+    ngx_snprintf((u_char *)err_msg, 256,
+                 "Invalid number of arguments. Expected exactly two: a key and "
+                 "a value.");
+    return err_msg;
+  }
+
+  ngx_str_t *values = (ngx_str_t *)(cf->args->elts);
+
+  std::string_view key = datadog::nginx::to_string_view(values[0]);
+  if (key.empty()) {
+    char *err_msg =
+        static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
+    ngx_snprintf((u_char *)err_msg, 256, "Empty key");
+    return err_msg;
+  }
+
+  std::string_view value = datadog::nginx::to_string_view(values[1]);
+  if (value.empty()) {
+    char *err_msg =
+        static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
+    ngx_snprintf((u_char *)err_msg, 256, "Empty value");
+    return err_msg;
+  }
+
+  (*rum_config)[std::string(key)] = std::string(value);
+  return NGX_CONF_OK;
+}
+
+static std::string make_rum_json_config(
+    std::string_view config_version,
+    const std::unordered_map<std::string, std::string> &config) {
+  rapidjson::Document doc;
+  doc.SetObject();
+
+  rapidjson::Document::AllocatorType &allocator = doc.GetAllocator();
+  doc.AddMember("majorVersion",
+                rapidjson::Value(std::stoi(config_version.data() + 1)),
```

Comment:
Edge case, but I think this will break if the config_version is just `v`

Reply:
This will definitely throw an uncatch exception. Good catch (no pun intended).
- [ ] sanitize args input.